### ISSUE TITLE
Can no longer only deploy firestore rules or firestore indexes

### ISSUE THREAD

**Issue by Burgov** [firebase/firebase-tools#4698]
### [REQUIRED] Environment info

**firebase-tools:** 11.2.0

**Platform:** any

### [REQUIRED] Test case

Just try to run the command

### [REQUIRED] Steps to reproduce

Run `firestore deploy --only firestore:indexes`

### [REQUIRED] Expected behavior

Only the firestore indexes would be deployed (and not the rules)

### [REQUIRED] Actual behavior

An error:

> Filters specified with colons (e.g. --only functions:func1,functions:func2) are only supported for functions and hosting

This change was introduced in https://github.com/firebase/firebase-tools/pull/4666 and I'm just wondering if this was intentional, since the commit message doesn't say anything about changing this behavior.

Until this commit, I was able to deploy only the rules or only the index (or both), but now I'm no longer able to do so.

Is there a reason behind this change, or was this an accidental side effect of a cleanup action?

**Comment by evanharmon1** [comment]
We appear to be having this issue as well. I believe it worked for us just a couple days ago and now we are getting the `Filters specified with colons (e.g. --only functions:func1,functions:func2) are only supported for functions and hosting` error

**Comment by hasanaktas** [comment]
Are there any developments?
Firebase version: 11.2.0

I just get error trying to deploy Firestore Security rules

**Comment by sivaneshl** [comment]
Looks like the syntax is changed. 

Deploy firestore rules 
`firebase deploy --only firestore:rules`

Deploy storage rules
`firebase deploy --only storage`

Here is the doc: 
https://firebase.google.com/docs/rules/manage-deploy#cloud-storage_1

**Comment by Burgov** [comment]
@sivaneshl that's what _used_ to work, but now I get this output:

![Screenshot 2022-07-04 at 08 50 09](https://user-images.githubusercontent.com/417674/177097706-fb9d76dc-93e0-4bc6-b13b-b4d6f31b7c3e.png)

**Comment by interstates21** [comment]
Same issue

**Comment by inaspic** [comment]
"firebase deploy --only firestore:rules" works with firebase-tools version 11.1.0, but not with version 11.2.0. Additional info...

**Comment by ryanhornberger** [comment]
just want to add a voice that our company is seeing the same problem on 11.2

**Comment by jessety** [comment]
+1, we've experiencing this as well. 11.1.0 seems to still work.

**Comment by bkendall** [comment]
This one might be me. The filters/checks for `only` and `except` were rather broken and I may have made them more strict than I intended. I'll take a look today.

(read: I didn't realize we supported `firestore:[rules|indexes]` - and that's very neat.)

**Comment by Andrew-Pynch** [comment]
We are experiencing this as well

**Comment by LanderBeeuwsaert** [comment]
jep, same, our CI is crashing on this

**Comment by hiaux0** [comment]
As found out by others above, versions prior to `11.2.0` are still working.
For us the fix was to pin the version in our `deployFirebase.yml` for the CI pipeline

```yml
    - name: Install Firebase CLI
      run: npm install -g firebase-tools@10.2.2     // <--- pin to 10.2.2 in our case
```

**Comment by behnamsattar** [comment]
Yayyyy, @bkendall thank you! 🎉 🙏

**Comment by phillipjacobs** [comment]
Still having the same issue as all the rest.
While waiting for a fix I just use `firebase deploy` or `npm install -g firebase-tools@10.2.2` for a temp solution

**Comment by RaheutRahwana** [comment]
11.2.1, the problem persists. Meanwhile `firebase deploy --only firestore` does the job. But be careful with the index.

**Comment by mckrava** [comment]
Hi there!
But how can I deal with such stuff like this:
`firebase deploy --only storage,firestore,firestore:rules,firestore:indexes`
It worked before `11.2.1` and I guess downgrade is not good idea.

**Comment by Burgov** [comment]
@mckrava `firestore` is shorthand for `firestore:rules,firestore:indexes`, so in your command it's redundant. You can simply do

```
firebase deploy --only storage,firestore
```

**Comment by yanoyuki** [comment]
I had the same problem with 11.2.1, but 11.2.2 works fine.

**Comment by bkendall** [comment]
This should be addressed in `11.2.2` - there was an issue with the parsing of `firestore:*`.
